agents-api/tests/test_execution_queries.py

```diff
@@ -443,3 +444,57 @@ async def _(
     assert result.status == "failed"
     assert result.error == "Something went wrong"
     assert result.transition_count == 2  # init + error
+
+
+@test("query: count transitions by developer")
```

Comment:
maybe we can start a paused execution, artificially increase transition count using sql on db_conn and then try to resume task? then it should fail, it is important to check that this limit works